phoenix-core/src/main/java/org/apache/phoenix/mapreduce/util/IndexColumnNames.java

```diff
@@ -61,8 +67,13 @@ public IndexColumnNames(final PTable pdataTable, final PTable pindexTable) {
         if (pindexTable.getViewIndexId() != null) {
             offset++;
         }
-        if (pindexTable.isMultiTenant() && pindexTable.getViewIndexId() != null) {
-            offset++;
+
+        if (tenantId != null) {
+            // For multi tenant view indexes, in order for skipscan to be used, we need tenant column.
+            // For tenant connections this column is not visible.
+            if (pindexTable.isMultiTenant() && pindexTable.getViewIndexId() != null) {
```

Comment:
@gokceni  Wouldn't you also need this for multi-tenant global indexes not on a view?